### PR TITLE
[ML][Pipelines]fix: mock_component_hash will break hash_dict logic in playback mode

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

Previous implementation of `mock_component_hash` will mock `hash_dict` and return a constant fake value, which
will break any logic depending on a valid component hash (e.g., in-memory cache for component registration) in playback mode.

In this PR, we have updated the implementation of `mock_component_hash` to return a valid hash value for `hash_dict` and updated all related recordings.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/schedule/e2etests/test_schedule.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 @pytest.mark.e2etest
 @pytest.mark.usefixtures("recorded_test", "mock_code_hash", "mock_asset_name", "mock_component_hash")
 @pytest.mark.pipeline_test
+@pytest.mark.mock_component_test
```
Comment (D-W-) [inline sdk/ml/azure-ai-ml/tests/schedule/e2etests/test_schedule.py:19]: What's this mark used for?

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/schedule/e2etests/test_schedule.py:19]: This is to rerun related tests, should be removed

File: sdk/ml/azure-ai-ml/tests/conftest.py [hunk 2]
```diff
@@ -487,17 +488,51 @@ def generate_uuid(*args, **kwargs):
...
 
     def generate_compononent_hash(*args, **kwargs):
         dict_hash = hash_dict(*args, **kwargs)
-        add_general_string_sanitizer(value=fake_component_hash, target=dict_hash)
+        normalized_dict_hash = normalized_hash_dict(*args, **kwargs)
+        add_general_string_sanitizer(value=normalized_dict_hash, target=dict_hash)
         return dict_hash
 
     if is_live():
         mocker.patch("azure.ai.ml.entities._component.component.hash_dict", side_effect=generate_compononent_hash)
```
Comment (D-W-) [inline sdk/ml/azure-ai-ml/tests/conftest.py:535]: In my understanding, in reply mode, we'll use actual component hash, how does the component code id match with the mock id in recoding?

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/conftest.py:535]: In live mode, the component dict will be like this when calculating anonymous hash:
```json
{
   ...
   "code": "azureml:/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/.../codes/xxx/versions/xxx"
   ...
}
```
while in playback mode, the returned arm id of code is from recording and has already been sanitized:
```json
{
   ...
   "code": "azureml:/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/.../codes/xxx/versions/xxx"
   ...
}
```
Suppose that we get `hash_a` in live mode and `hash_b` in playback mode, here comes the mismatch: recorded requests are based on `hash_a` (live result) but we will use `hash_b` for requests in playback mode. In this PR, we replace all `hash_a` in recording with `hash_b`, which is aligned with playback mode behavior

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/conftest.py:535]: In previous solution, we change both recording and hash behavior in playback mode; in this PR, we change recording only.

**Review by brynn-code — APPROVED** [review]

**Pull request merged**